gcc/rust/checks/errors/rust-readonly-check2.h

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2025 Free Software Foundation, Inc.
+
+// This file is part of GCC.
+
+// GCC is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3, or (at your option) any later
+// version.
+
+// GCC is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with GCC; see the file COPYING3.  If not see
+// <http://www.gnu.org/licenses/>.
+
+#include "rust-hir-visitor.h"
+#include "rust-name-resolver.h"
+#include "rust-stacked-contexts.h"
+#include "rust-hir-type-check.h"
+
+namespace Rust {
+namespace HIR {
+class ReadonlyChecker : public DefaultHIRVisitor
+{
+public:
+  ReadonlyChecker ();
+
+  void go (HIR::Crate &crate);
+
+private:
+  enum lvalue_use
+  {
+    lv_assign,
+    lv_increment,
+    lv_decrement,
+  };
```

Comment:
```suggestion
  enum case lvalue_use
  {
    assign,
    increment,
    decrement,
  };
```

this way we get a bit stronger namespacing

Reply:
oops, meant enum class thank you @P-E-P